File: MinVid/src/app/Models/videoMetadata.ts
export class VideoMetadata {
    id: string;
    title: string;
    format: string;
    tags: string[]; 
    description: string;
    uploadDate: Date;
    duration: number; 
    isShort?: boolean;
    is360?: boolean;

    constructor(title: string, format: string, description: string, tags: string[], uploadDate: Date, duration: number, isShort?: boolean, is360?: boolean) {
        this.title = title;
        this.format = format;
        this.description = description;
        this.tags = tags;
        this.uploadDate = new Date();
        this.duration = duration;
        this.isShort = isShort;
        this.is360 = is360;
    }
}

File: MinVid-API/Models/VideoMetadata.cs
namespace MinVid_API.Models
{
    public class VideoMetadata
    {
        public string id { get; set; }
        public string title { get; set; }
        public string format { get; set; }
        public string description { get; set; } 
        public List<string> tags { get; set; }
        public DateTime uploadDate { get; set; }
        public int duration { get; set; }
    }
}
